app/controllers/msa_components_controller.rb

```diff
@@ -4,20 +4,32 @@ def index
   end
 
   def new
-    @component = NewMsaComponentEvent.new
+    begin
+      authorize NewMsaComponentEvent
+      @component = NewMsaComponentEvent.new
+    rescue Pundit::NotAuthorizedError
+      flash[:warn] = "You are not authorised to do this action"
```

Comment:
any content/text strings should be now defined in the locale file

Reply:
have updated!